src/audio/utils.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "misc/slice.hpp"
+
+using InputBuffer = ConstSlice<float>;
+using OutputBuffer = Slice<float>;
+
+namespace AbaddonClient::Audio {
+
+class AudioUtils {
+public:
+    AudioUtils() = delete;
+    ~AudioUtils() = delete;
+
+    static void ApplyGain(OutputBuffer buffer, float gain) noexcept {
+        for (auto &sample : buffer) {
+            sample *= gain;
+        }
+    }
+
+    static void ClampToFloatRange(OutputBuffer buffer) noexcept {
+        for (auto& sample : buffer) {
+            sample = std::clamp(sample, -1.0f, 1.0f);
+        }
+    }
+
+    static void MixStereoToMono(OutputBuffer buffer) noexcept {
+        for (auto iter = buffer.begin(); iter < buffer.end() - 2; iter += 2) {
+            const auto mixed = std::reduce(iter, iter + 2, 0) / 2.0f;
```

Comment:
```suggestion
            const auto mixed = std::reduce(iter, iter + 2, 0.0f) / 2.0f;
```
reduce is deducing as int here so mix doesnt work. unrelated but it looks like discord is mixing down to mono on the receiving end too now. >:^(